python/cudf/cudf/tests/test_text.py

```diff
@@ -140,6 +140,39 @@ def test_ngrams(n, separator, expected_values):
     assert_series_equal(expected.to_pandas(), actual.to_pandas())
 
 
+@pytest.mark.parametrize(
+    "n, expected_values",
+    [
+        (
+            2,
+            [
+                "th",
+                "hi",
+                "is",
+                "is",
+                "my",
+                "bo",
+                "oo",
+                "ok",
+                "he",
+                "er",
+                "re",
+            ],
+        ),
+        (3, ["thi", "his", "boo", "ook", "her", "ere"]),
+    ],
+)
+def test_character_ngrams(n, expected_values):
```

Comment:
Curious about the behaviour when `n<=0`, or `n` is very large -- anything exciting?

Reply:
This function and other nvtext ngram generating functions only support `ngrams >= 2` (throws an exception).

The _n_ part of _ngrams_ is the number of adjacent items to join together to make a new item. They are used in NLP to create a corpus of phrase-like dictionaries for various reasons. The most common ngram is the bigram (n=2) but higher ngrams like trigrams (n=3) are also sometimes generated. I've not seen many ngrams higher than n=5 other than in academic text books. 

And so `n<=0` does not really make sense (hence the exception). Unigrams (n=1) are usually no-ops (nothing to combine). Here unigrams could be more efficiently created using the `nvtext::character_tokenizer()` instead of this call.

Reply:
Thanks! Nothing to change here, just a question!